### PR TITLE
chore(deps): Update typescript in components

### PR DESCRIPTION
**Pull request by eddysims**
## Motivations

We recently updated typescript in Atlantis, however missed updating it in the components package. This PR updates typescript in components.

While updating typescript we found a few type bugs in components that caused `npm ci` to fail. These bugs have also been resolved in this PR to allow `npm ci` to finish running.

## Changes

- Update typescript
- Update framer-motion

### Fixed

- Type declaration on InputText and FormField

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/FeatureSwitch/FeatureSwitch.tsx [hunk 1]
```diff
@@ -109,7 +109,6 @@ export function FeatureSwitch({
                 transition={{
                   delay: 0.2,
                   type: "spring",
-                  duration: 0.2,
```
Comment (eddysims) [inline packages/components/src/FeatureSwitch/FeatureSwitch.tsx:112]: We are able to remove these durations as they were not used.

When we look at that [docs](https://www.framer.com/api/motion/types/#spring) for the "spring" type, it does not accept a duration option.

**Comment by JeffMarvin** [comment]
@eddysims 

From the PR description: 
> ... and makes a few fixes to components along with it.

There are some type fixes here, but ideally this would be a good thing to separate out into its own PR. Currently, there is no reasoning or description as to why the type/component changes are being made. So, it's hard to review/QA this.

Could you add a bit more context (either inline in comments or in the PR description) to describe why the other changes in this PR are being made?

File: packages/components/src/FormField/FormField.tsx [hunk 2]
```diff
@@ -178,10 +178,7 @@ export const FormField = React.forwardRef(
       value,
       validations,
     }: FormFieldProps,
-    ref:
-      | Ref<HTMLInputElement>
-      | Ref<HTMLTextAreaElement>
-      | Ref<HTMLSelectElement>,
+    ref: Ref<HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement>,
```
Comment (eddysims) [inline packages/components/src/FormField/FormField.tsx:181]: This was written incorrectly and was causing issues within the `InputText` component.

**Comment by JeffMarvin** [comment]
LGTM, thanks for the additional context on the type changes. 
This passed a quick smoke check (rendered the docs pages for the affected components).

**Pull request merged**